### PR TITLE
.net6 and vs2022 fixes as per #25341

### LINKED ISSUE #25341: .NET 6: Publish an ASP.NET Core web API to Azure API Management with VS
Replace "At this point, you need to add an API to the Azure API Management service. Leave Visual Studio open while you complete the following tasks." creating the API in the current dialog. It now support that.

Remove all the pic's. The pic's are so big they're not helpful anyway and change too often.

Copy/paste the beginning instructions from https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api
have them  Delete `[Route("[controller]")] ` and add `[Route("/")]`

In the instructions
Replace `services.AddSwaggerGen();` with
```
services.AddSwaggerGen(config =>
{
    config.SwaggerDoc("v1", new Microsoft.OpenApi.Models.OpenApiInfo
    {
        Title = "WeatherForecasts",
        Version = "v1"
    });
});
```
.NET 3.2 and 5.0 both generated the replacement code, not so in .NET 6

> Notice the name of the API is different than what you named it
I don't think that's true when you create it from the pub profle.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d70145c2-e2ba-77fb-050d-44d18354e370
* Version Independent ID: 19c98d91-e5fe-5832-6c75-23e677e469a3
* Content: [Publish an ASP.NET Core web API to Azure API Management with Visual Studio](https://docs.microsoft.com/en-us/aspnet/core/tutorials/publish-to-azure-api-management-using-vs?view=aspnetcore-6.0)
* Content Source: [aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @codemillmatt
* Microsoft Alias: **masoucou**

### PR DESCRIPTION
**Pull request by codemillmatt**
Fixes #25341

### CODE REVIEW

**Comment by codemillmatt** [comment]
@Rick-Anderson  - FYI

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 1]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+* Open a [free Azure account](https://azure.microsoft.com/free/dotnet/) if you don't have one.
 
-Adding Swagger definitions to the ASP.NET Core web API allows Azure API Management to read the app's API definitions. Complete the following steps.
+## Create an ASP.NET Core web API
 
-### Add Swagger
+Visual Studio allows you to easily create a new ASP.NET Core web API project from a template. Follow these directions to create a new ASP.NET Core web API project:
 
-1. Add the [Swashbuckle.AspNetCore](https://www.nuget.org/packages/Swashbuckle.AspNetCore) NuGet package to the ASP.NET Core web API project:
+* From the File menu, select New > Project.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:26]:
```suggestion
* From the File menu, select **New** > **Project**.
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 2]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+* Enter Web API in the search box.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:27]:
```suggestion
* Enter **Web API** in the search box.
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 3]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+* Select the ASP.NET Core Web API template and select Next.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:28]:
```suggestion
* Select the **ASP.NET Core Web API** template and select **Next**.
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 4]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+* In the Configure your new project dialog, name the project WeatherAPI and select Next.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:29]:
```suggestion
* In the **Configure your new project dialog**, name the project **WeatherAPI** and select **Next**.
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 5]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+* In the Additional information dialog:
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:30]:
```suggestion
* In the **Additional information** dialog:
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 6]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+* Confirm the Framework is .NET 6.0 (Long-term support).
+* Confirm the checkbox for Use controllers (uncheck to use minimal APIs) is checked.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:32]:
```suggestion
* Confirm the checkbox for **Use controllers (uncheck to use minimal APIs)** is checked.
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 7]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
-### Add Swagger
+Visual Studio allows you to easily create a new ASP.NET Core web API project from a template. Follow these directions to create a new ASP.NET Core web API project:
 
-1. Add the [Swashbuckle.AspNetCore](https://www.nuget.org/packages/Swashbuckle.AspNetCore) NuGet package to the ASP.NET Core web API project:
+* From the File menu, select New > Project.
+* Enter Web API in the search box.
+* Select the ASP.NET Core Web API template and select Next.
+* In the Configure your new project dialog, name the project WeatherAPI and select Next.
+* In the Additional information dialog:
+* Confirm the Framework is .NET 6.0 (Long-term support).
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:31]:
```suggestion
* Confirm the Framework is **.NET 6.0 (Long-term support)**.
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 8]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+* Confirm the checkbox for Use controllers (uncheck to use minimal APIs) is checked.
+* Confirm the checkbox for Enable OpenAPI support is checked.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:33]:
```suggestion
* Confirm the checkbox for **Enable OpenAPI support** is checked.
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 9]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+* Select Create.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:34]:
```suggestion
* Select **Create**.
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 10]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+## Explore the code
 
-1. Add the following line to the `Startup.ConfigureServices` method:
-    
-    ```csharp
-    services.AddSwaggerGen();
-    ```
-    
-1. Add the following line to the `Startup.Configure` method:
+Swagger definitions allow Azure API Management to read the app's API definitions. By checking the _Enable OpenAPI support_ checkbox during app creation, Visual Studio will automatically add the code to create the Swagger definitions. Open up the `Program.cs` file and you'll see the following code:
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:38]:
```suggestion
Swagger definitions allow Azure API Management to read the app's API definitions. By checking the **Enable OpenAPI support** checkbox during app creation, Visual Studio automatically adds the code to create the Swagger definitions. Open up the `Program.cs` file which shows the following code:
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 11]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+
+...
+
+```
+
+### Ensure the Swagger definitions are always generated
+
+Azure API Management needs the Swagger definitions to always be present, regardless of the application's environment. To ensure they are always generated, move `app.UseSwagger();` outside of the `if (app.Environment.IsDevelopment())` block. In addition, you can 
+
+The code should now look like this:
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:62]:
```suggestion
The updated code:
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 12]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+    app.UseSwaggerUI();
+}
+
+...
+
+```
+
+### Ensure the Swagger definitions are always generated
+
+Azure API Management needs the Swagger definitions to always be present, regardless of the application's environment. To ensure they are always generated, move `app.UseSwagger();` outside of the `if (app.Environment.IsDevelopment())` block. In addition, you can
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:60]: > In addition, you can
can what?

BTW, in this repo we prefer to not use `you`. It's generally unnecessary.

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 13]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
+    app.UseSwaggerUI();
+}
+
+...
+
+```
 
 ### Change the API routing
 
 Next, you'll change the URL structure needed to access the `Get` action of the `WeatherForecastController`. Complete the following steps:
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:81]:
```suggestion
Change the URL structure needed to access the `Get` action of the `WeatherForecastController`. Complete the following steps:
```
Next is dead wood and banned from this repo.

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 14]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
 
 1. Open the `WeatherForecastController.cs` file.
-1. Delete the `[Route("[controller]")]` class-level attribute. The class definition will look like the following:
+1. Replace the `[Route("[controller]")]` class-level attribute with `[Route("/")]`. The class definition will look like the following:
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:84]:
```suggestion
1. Replace the `[Route("[controller]")]` class-level attribute with `[Route("/")]`. The updated class definition :
```

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 15]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
-    ![Screenshot of the newly created Azure API Management blank API selected in the publish dialog](publish-to-azure-api-management-using-vs/_static/publish_dialog_api_selected.png)
+    Select the **OK** button to create the API Management service.
 
+1. Select the **Create** button to proceed with the API Management service creation. This step may take several minutes to complete.
+1. When that completes, select the **Finish** button.
 1. The dialog closes and a summary screen appears with information about the publish. Select the **Publish** button.
 
-    ![Screenshot of Visual Studio with the publish profile displayed](publish-to-azure-api-management-using-vs/_static/vs_publish_profile.png)
-
     The web API will publish to both Azure App Service and Azure API Management. A new browser window will appear and show the API running in Azure App Service. You can close that window.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:122]:
```suggestion
    The web API publishs to both Azure App Service and Azure API Management. A new browser window will appear and show the API running in Azure App Service. You can close that window.
```
Sorry, fixing style guide violation. Future tense is prohibited unless needed.

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md [hunk 16]
```diff
@@ -4,192 +4,147 @@ author: codemillmatt
...
-
-1. Open the newly created publish profile. It can be found from **Solution Explorer** in the *Properties/PublishProfiles* folder.
-
-    ![Screenshot showing the publish profile file location highlighted](publish-to-azure-api-management-using-vs/_static/vs_publish_profile_highlighted.png)
-
-1. Change the `<OpenAPIDocumentName>` element's value from `v1` to `WeatherForecasts`.
-
-    ![screenshot of the changes necessary for the publish profile](publish-to-azure-api-management-using-vs/_static/vs_publish_profile_changes.png)
-
+
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:144]: All of this is really clumsy. Can you open an issue on this doc and I'll create the project and import snippets. With snippets you can show more code and highlight the updates, changes, etc.

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md (hunk 16 above)
Comment (codemillmatt) [reply aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md:144]: Yes, I'll create the issue after the merge.

File: aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md (hunk 12 above)
Comment (codemillmatt) [reply aspnetcore/tutorials/publish-to-azure-api-management-using-vs.md]: Updated in subsequent commit.

**Review by Rick-Anderson — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
@codemillmatt much appreciated. It should be live in 20 minutes.

**Pull request merged**